src/PointwiseFunctions/AnalyticSolutions/Elasticity/Zero.hpp

```diff
@@ -0,0 +1,106 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <pup.h>
+
+#include "DataStructures/DataBox/Prefixes.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Elliptic/Systems/Elasticity/Tags.hpp"
+#include "Options/Options.hpp"
+#include "Parallel/CharmPupable.hpp"
+#include "PointwiseFunctions/AnalyticData/AnalyticData.hpp"
+#include "Utilities/MakeWithValue.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+/// \cond
+class DataVector;
+namespace PUP {
+class er;
+}  // namespace PUP
+/// \endcond
+
+namespace Elasticity::Solutions {
+
+/// \cond
+template <size_t Dim, typename Registrars>
+struct Zero;
+
+namespace Registrars {
+template <size_t Dim>
+struct Zero {
+  template <typename Registrars>
+  using f = Solutions::Zero<Dim, Registrars>;
+};
+}  // namespace Registrars
```

Comment:
(Not a change request.)  Classes with a `size_t Dim` parameter and then (possibly) type parameters seem very common.  Do people think it would be worth having a registrar helper for that case like we do for the all-type case?

Reply:
Actually I found it slightly confusing to have the other registrar helper because it took me a while to realize that it just abbreviates the no-other-template-params case. Possibly easier to just remove the helpers and have people write the registrar. No strong opinion here.

Reply:
I think you're still confused, then.  It abbreviates the no-non-type-template-parameters case.

The reason I don't like making people write them is that they are actually a pretty advanced TMP pattern (although admittedly a very simple case of it).  Also I'm lazy and don't like writing boilerplate.